controller/backup_target_controller.go

```diff
@@ -546,22 +553,48 @@ func (btc *BackupTargetController) getInfoFromBackupStore(backupTarget *longhorn
 	return info, nil
 }
 
-func (btc *BackupTargetController) syncBackupVolume(backupTarget *longhorn.BackupTarget, backupStoreBackupVolumeNames []string, syncTime metav1.Time, log logrus.FieldLogger) error {
-	backupStoreBackupVolumes := sets.New[string](backupStoreBackupVolumeNames...)
-
+func (btc *BackupTargetController) cleanupDuplicateBackupVolumes(backupTargetName string, log logrus.FieldLogger) (map[string]*longhorn.BackupVolume, error) {
 	// Get a list of all the backup volumes that exist as custom resources in the cluster
-	clusterBackupVolumes, err := btc.ds.ListBackupVolumesWithBackupTargetNameRO(backupTarget.Name)
+	clusterBackupVolumes, err := btc.ds.ListBackupVolumesWithBackupTargetNameRO(backupTargetName)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
+	deletingDuplicateBackupVolumeSet := sets.New[string]()
 	clusterVolumeBVMap := make(map[string]*longhorn.BackupVolume, len(clusterBackupVolumes))
-	clusterBackupVolumesSet := sets.New[string]()
 	for _, bv := range clusterBackupVolumes {
-		clusterBackupVolumesSet.Insert(bv.Spec.VolumeName)
+		if bv.Spec.VolumeName == "" || bv.Spec.BackupTargetName == "" || bv.Spec.BackupTargetName != backupTargetName {
+			log.WithField("backupVolume", bv.Name).Debugf("Volume name or backup target name is empty or different backup target name %v in BackupVolume, deleting it", bv.Spec.BackupTargetName)
```

Comment:
Need to refactor the if codes and make the log messages more granular to improve the troubleshooting capability.
```suggestion
		if bv.Spec.VolumeName == "" || bv.Spec.BackupTargetName == "" || bv.Spec.BackupTargetName != backupTargetName {
			log.WithField("backupVolume", bv.Name).Warn("Found volume name is empty") bv.Spec.BackupTargetName)
		} elseif bv.Spec.BackupTargetName == "" {
			log.WithField("backupVolume", bv.Name).Warn("Found volume name is empty") 
		} else if bv.Spec.BackupTargetName != backupTargetName {
			log....
		}
```

Reply:
Modified. Thanks.